pallets/pips/src/lib.rs

```diff
@@ -1490,21 +1490,19 @@ impl<T: Trait> Module<T> {
     fn adjust_live_queue(id: PipId, old: SnapshottedPip<BalanceOf<T>>) {
         let new = Self::aggregate_result(id);
         <LiveQueue<T>>::mutate(|queue| {
-            let old_pos = queue.binary_search_by(|res| compare_spip(res, &old));
-            let new_pos = queue.binary_search_by(|res| compare_spip(res, &new));
-            match (old_pos, new_pos) {
-                // First time adding to queue, so insert.
-                (Err(_), Ok(pos) | Err(pos)) => queue.insert(pos, new),
-                // Already in queue. Swap positions.
-                (Ok(old_pos), Err(new_pos)) => {
-                    // Cannot underflow by definition of binary search finding an element.
-                    let new_pos = new_pos.min(queue.len() - 1);
-                    move_element(queue, old_pos, new_pos);
-                    queue[new_pos] = new;
-                }
-                // We have `old_res == new_res`. Queue state is already good.
-                (Ok(_), Ok(_)) => {}
-            }
+            // Remove the old element. We can assume its in the list and findable,
+            // as the list is sorted, updated, and old is taken before modification.
+            let old_pos = queue
+                .binary_search_by(|res| compare_spip(res, &old))
+                .unwrap();
+            queue.remove(old_pos);
```

Comment:
Let's not assume that the old element is present in the list.
```suggestion
            // Remove the old element.
            if let Ok(old_pos) = queue.binary_search_by(|res| compare_spip(res, &old)) {
                queue.remove(old_pos);
            }          
```